src/pages/gl-segments/index.js

```diff
@@ -0,0 +1,198 @@
+      })
+
+      setData(response)
+    } else {
+      setData([])
+    }
+  }
+
+  const {
+    labels: _labels,
+    paginationParameters,
+    invalidate,
+    refetch,
+    access
+  } = useResourceQuery({
+    queryFn: fetchGridData,
+    endpointId: GeneralLedgerRepository.Segments.qry,
+    datasetId: ResourceIds.Segments
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      segmentName: null
+    },
+    maxAccess: access,
+    enableReinitialize: true,
+    validateOnChange: true
+  })
+
+  useEffect(() => {
+    fetchGridData()
+  }, [formik.values.segmentId])
+
+  const columns = [
+    {
+      field: 'reference',
+      headerName: _labels.reference,
+      flex: 1
+    },
+    {
+      field: 'name',
+      headerName: _labels.segment,
+      flex: 1
+    }
+  ]
+
+  const del = async obj => {
+    try {
+      await postRequest({
+        extension: GeneralLedgerRepository.Segments.del,
+        record: JSON.stringify(obj)
+      })
+      invalidate()
+      toast.success(platformLabels.Deleted)
+    } catch (error) {}
+  }
+
+  const edit = obj => {
+    openForm(obj)
+  }
+
+  const add = () => {
+    openForm()
+  }
+
+  function openForm(obj) {
+    stack({
+      Component: SegmentForm,
+      props: {
+        labels: _labels,
+        obj,
+        maxAccess: access,
+        formikSegmentId: formik.values.segmentId,
+        fetchGridData
+      },
+      width: 500,
+      height: 260,
+      title: _labels.segment
+    })
+  }
+
+  useEffect(() => {
+    ;(async function () {
+      const res = await getRequest({
+        extension: SystemRepository.Defaults.qry,
+        parameters: '_filter=GLACSeg'
+      })
+
+      if (res && res.list) {
+        const filteredList = res.list
+          .filter(item => item.key.startsWith('GLACSegName') && item.value !== null)
+          .map(item => {
+            const segKey = item.key.replace('Name', '')
+            const matchingSeg = res.list.find(seg => seg.key === segKey && seg.value !== null)
+            if (matchingSeg) {
+              return {
+                key: item.value,
+                value: matchingSeg.value
+              }
+            }
+
+            return null
+          })
+          .filter(Boolean)
+
+        setStore(filteredList)
+
+        if (filteredList.length > 0) {
+          formik.setFieldValue('segmentName', filteredList[0].key)
+          formik.setFieldValue('segmentId', filteredList[0].value)
+        }
+      }
+    })()
+  }, [])
+
+  return (
+    <VertLayout>
+      <Fixed>
+        <GridToolbar
+          onAdd={add}
+          maxAccess={access}
+          labels={_labels}
+          middleSection={
+            <Grid item sx={{ display: 'flex', mr: 2 }}>
+              <CustomComboBox
+                endpointId={SystemRepository.Defaults.qry}
+                parameters={`_filter=GLACSeg`}
+                sx={{ width: 450 }}
```

Comment:
i the styling here needed?

Reply:
![image](https://github.com/user-attachments/assets/70b6bebb-ead2-44cf-ab94-825cbff0f40a)
both ways we need it smaller

Reply:
yes 
![image](https://github.com/user-attachments/assets/c9294f1c-e8bf-4e2c-b78c-9f46250d554e)